### PR TITLE
Update queue flags to be optional

### PR DESCRIPTION
**Pull request by noelatmerqury**
Set the `Queue` flags to be optional because they may not always be provided. For example running the below commands would result in an output `JSON` that would cause the library to return an error when parsing.

```bash
sudo nft add table ip mytable
sudo nft add chain ip mytable out-chain "{ type filter hook output priority 0;}"
sudo nft add rule ip mytable out-chain udp dport 20000 queue num 0
```

This `PR` fixes this issue by setting the queue flags as optional. An associated unit test has been added to verify functionality.

### CODE REVIEW

**Comment by noelatmerqury** [comment]
Fixing an issue with serialisation to JSON. Will push a fix soon.

**Comment by noelatmerqury** [comment]
> Fixing an issue with serialisation to JSON. Will push a fix soon.

@JKRhb This has now been fixed and tested.

**Review by JKRhb — COMMENTED** [review]
Thank you for your PR, @noelatmerqury :) The changes look good to me, we'll be able to merge as soon as @jwhb has given his approval :)

**Comment by jwhb** [comment]
Nice contribution, thanks @noelatmerqury!
LGTM.

**Review by jwhb — APPROVED** [review]

**Pull request merged**
